### PR TITLE
feat(swap): display warning for safe token

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

It is expected that SAFE token will have a lot of volatility on their transferability event. 

This PR adds a warning to make users aware of this, and to encourage them to review their slippage tolerance settings.

- [x] Feature flag. Already enabled
- [x] Banner
- [x] Show when SAFE is selected

image

## Test 
Only for SWAP. 

- Notice the warning shows when you select SAFE token as a sell or buy token
- If you dismiss the message, should not show it again

### CODE REVIEW

**Review by elena-zh — APPROVED** [review]
LGTM

**Comment by elena-zh** [comment]
Maybe just a nitpick is to show this message not only when Safe token is selected, but additionally an amount to trade is specified. WDYT?

File: apps/cowswap-frontend/src/modules/swap/containers/SafeTokenBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  buyTokenAddress: string | undefined
+}
+
+export function SafeTokenBanner({ sellTokenAddress, buyTokenAddress }: SafeTokenBanner) {
+  const { showSafeTokenWarning } = useFeatureFlags()
+
+  const isSafeTokenSelected =
+    sellTokenAddress?.toLowerCase() === SAFE_TOKEN || buyTokenAddress?.toLowerCase() === SAFE_TOKEN
+
+  console.log('isSafeTokenSelected', { sellTokenAddress, buyTokenAddress, isSafeTokenSelected })
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/swap/containers/SafeTokenBanner.tsx:21]: To console.log should be deleted

**Review by shoom3301 — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/swap/containers/SafeTokenBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import BRAIN_IMAGE from '@cowprotocol/assets/svg/brain.svg'
+import { useFeatureFlags } from '@cowprotocol/common-hooks'
+import { BannerOrientation, ClosableBanner, InlineBanner } from '@cowprotocol/ui'
+
+import { Settings } from 'react-feather'
+
+const SAFE_TOKEN = '0x5afe3855358e112b5647b952709e6165e1c1eeee'
+const SAFE_TOKEN_WARNING_STORAGE_KEY = 'safeTokenWarning:v0'
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/swap/containers/SafeTokenBanner.tsx:8]: Should we have this warning for the token on gnosis chain too?

Or, should we still have this warning now that the first day is passed?

Comment (anxolin) [reply apps/cowswap-frontend/src/modules/swap/containers/SafeTokenBanner.tsx:8]: Good question :) 

I think we should remove. I will ask around

Comment (anxolin) [reply apps/cowswap-frontend/src/modules/swap/containers/SafeTokenBanner.tsx:8]: done

**Pull request merged**